### PR TITLE
[WebSocket] Add support for Continuation WebSocket Frames

### PR DESCRIPTION
**Pull request by irunika**
This resolves https://github.com/wso2/transport-http/issues/135

### CODE REVIEW

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketSourceHandler.java [hunk 1]
```diff
@@ -155,27 +159,50 @@ public void channelRead(ChannelHandlerContext ctx, Object msg)
             logger.error("Expecting WebSocketFrame. Unknown type.");
             throw new UnknownWebSocketFrameTypeException("Expecting WebSocketFrame. Unknown type.");
         }
+
+        // If the continuation of frames are not following the protocol, netty handles it internally.
+        // Because of that those situations are not handles here.
```
Comment (riyafa) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketSourceHandler.java]: Shall we change?
```
// If the continuation of frames are not following the protocol, netty handles them internally.
 // Hence those situations are not handled here.
```

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/sender/websocket/WebSocketTargetHandler.java [hunk 2]
```diff
@@ -147,11 +151,22 @@ public void channelRead(ChannelHandlerContext ctx, Object msg)
                     "Unexpected FullHttpResponse (getStatus=" + response.status() +
                             ", content=" + response.content().toString(CharsetUtil.UTF_8) + ')');
         }
+
+        // If the continuation of frames are not following the protocol, netty handles it internally.
+        // Because of that those situations are not handles here.
```
Comment (riyafa) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/sender/websocket/WebSocketTargetHandler.java]: same here update comment

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketSourceHandler.java [hunk 3]
```diff
@@ -155,27 +159,50 @@ public void channelRead(ChannelHandlerContext ctx, Object msg)
...
         if (msg instanceof TextWebSocketFrame) {
-            notifyTextMessage((TextWebSocketFrame) msg);
+            TextWebSocketFrame textFrame = (TextWebSocketFrame) msg;
+            if (!textFrame.isFinalFragment()) {
+                continuationFrameType = WebSocketFrameType.TEXT;
+            }
+            notifyTextMessage(textFrame, textFrame.text(), textFrame.isFinalFragment());
```
Comment (riyafa) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketSourceHandler.java]: Only trying to understand. Will the `TextWebSocketFrame` come before a `ContinuationWebSocketFrame` ? Then do we first set the `continuationFrameType` and then notify it and then notify the continuationframes afterwards?

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketSourceHandler.java (hunk 1 above)
Comment (irunika) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketSourceHandler.java]: To what?

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/sender/websocket/WebSocketTargetHandler.java (hunk 2 above)
Comment (irunika) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/sender/websocket/WebSocketTargetHandler.java]: To what?

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketSourceHandler.java (hunk 3 above)
Comment (irunika) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketSourceHandler.java]: It does not come every time. 
If WebSocketFrame -> isFinalFragment = false 
then next frame should be a continuation frame. Errors of this sequence are already handled by netty in transport level. 

I actually cannot understand your suggestion. Next frame will not hit the channel read until one frame processing is finished.

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketSourceHandler.java (hunk 1 above)
Comment (riyafa) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketSourceHandler.java]: to this: 
```
// If the continuation of frames are not following the protocol, netty handles them internally.
 // Hence those situations are not handled here.
```

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketSourceHandler.java (hunk 3 above)
Comment (riyafa) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketSourceHandler.java]: Got it thanks

Comment (irunika) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketSourceHandler.java]: :)

**Pull request merged**